vm/string.go

```diff
@@ -471,6 +472,719 @@ func builtinStringInstanceMethods() []*BuiltInMethodObject {
 				}
 			},
 		},
+		{
+			// Returns a string which is concatenate with the input string or character
+			//
+			// ```ruby
+			// "Hello ".concat("World") # => "Hello World"
+			// ```
+			// @return [String]
+			Name: "concat",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+
+					str := receiver.(*StringObject).Value
+					concatStr, ok := args[0].(*StringObject)
+
+					if !ok {
+						return wrongTypeError(receiver.Class())
+					}
+
+					return t.vm.initStringObject(str + concatStr.Value)
+				}
+			},
+		},
+		{
+			// Returns the character of the string with specified index
+			// It will raise error if the input is not an Integer type
+			//
+			// ```ruby
+			// "Hello"[1]        # => "e"
+			// "Hello"[5]        # => nil
+			// "Hello\nWorld"[5] # => "\n"
+			// "Hello"[-1]       # => "o"
+			// "Hello"[-6]       # => nil
+			// ```
+			//
+			// @return [String]
+			Name: "[]",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+					if len(args) != 1 {
+						return initErrorObject(ArgumentErrorClass, "Expect 1 argument. got=%v", strconv.Itoa(len(args)))
+					}
+
+					str := receiver.(*StringObject).Value
+					i := args[0]
+					index, ok := i.(*IntegerObject)
+
+					if !ok {
+						return initErrorObject(TypeErrorClass, "Expect Integer. got=%T (%+v)", args[0], args[0])
```

Comment:
I'm afraid implementing rune type is a kind of future work, because this is a comprehensive issue on Goby and we need survey before doing that.